sources/Interop/D3D12MemoryAllocator/include/Allocation.cs

```diff
@@ -0,0 +1,409 @@
+            uint m_ResourceDimension
+            {
+                readonly get => (uint)BitHelper.ExtractRange(__value, 2, 3);
+                set => __value = BitHelper.SetRange(__value, 2, 3, value);
+            }
+
+            [NativeTypeName("UINT")]
+            uint m_ResourceFlags
+            {
+                readonly get => (uint)BitHelper.ExtractRange(__value, 5, 24);
+                set => __value = BitHelper.SetRange(__value, 5, 24, value);
+            }
+
+            [NativeTypeName("UINT")]
+            uint m_TextureLayout
+            {
+                readonly get => (uint)BitHelper.ExtractRange(__value, 29, 9);
+                set => __value = BitHelper.SetRange(__value, 29, 9, value);
+            }
+
+            [NativeTypeName("UINT")]
+            uint m_WasZeroInitialized
+            {
+                readonly get => (uint)BitHelper.ExtractRange(__value, 38, 1);
+                set => __value = BitHelper.SetRange(__value, 38, 1, value);
+            }
+        }
+
+        internal Allocation(AllocatorPimpl* allocator, [NativeTypeName("UINT64")] ulong size, [NativeTypeName("BOOL")] int wasZeroInitialized)
+        {
+            Unsafe.SkipInit(out this);
+
+            m_Allocator = allocator;
+            m_Size = size;
+            m_Resource = null;
+            m_CreationFrameIndex = allocator->GetCurrentFrameIndex();
+            m_Name = null;
+
+            D3D12MA_ASSERT(allocator != null);
+
+            m_PackedData.SetType(TYPE_COUNT);
+            m_PackedData.SetResourceDimension(D3D12_RESOURCE_DIMENSION_UNKNOWN);
+            m_PackedData.SetResourceFlags(D3D12_RESOURCE_FLAG_NONE);
+            m_PackedData.SetTextureLayout(D3D12_TEXTURE_LAYOUT_UNKNOWN);
+            m_PackedData.SetWasZeroInitialized(wasZeroInitialized);
+        }
+
+        public void Dispose()
+        {
+            // Nothing here, everything already done in Release.
+        }
+
+        internal void InitCommitted(D3D12_HEAP_TYPE heapType)
+        {
+            m_PackedData.SetType(TYPE_COMMITTED);
+            m_Union.m_Committed.heapType = heapType;
+        }
+
+        internal void InitPlaced([NativeTypeName("UINT64")] ulong offset, [NativeTypeName("UINT64")] ulong alignment, NormalBlock* block)
+        {
+            m_PackedData.SetType(TYPE_PLACED);
+            m_Union.m_Placed.offset = offset;
+            m_Union.m_Placed.block = block;
+        }
+
+        internal void InitHeap(D3D12_HEAP_TYPE heapType, ID3D12Heap* heap)
+        {
+            m_PackedData.SetType(TYPE_HEAP);
+            m_Union.m_Heap.heapType = heapType;
+            m_Union.m_Heap.heap = heap;
+        }
+
+        internal void SetResource<TD3D12_RESOURCE_DESC>(ID3D12Resource* resource, [NativeTypeName("const D3D12_RESOURCE_DESC_T*")] TD3D12_RESOURCE_DESC* pResourceDesc)
+            where TD3D12_RESOURCE_DESC : unmanaged
+        {
+            D3D12MA_ASSERT(m_Resource == null && pResourceDesc != null);
```

Comment:
nit: Parentheses help to visually parse the parts of the expression
```suggestion
            D3D12MA_ASSERT((m_Resource == null) && (pResourceDesc != null));
```

Reply:
Done in 6ad85a89365dbe6b7d6b0024fe904e56e715e596.